taskcoordinator/app/src/test/java/com/microservices/taskcoordinator/controllers/CoordinatorControllerIntegrationTests.java

```diff
@@ -74,15 +96,178 @@ public void testCoordinateOrder_allValid_orderCoordinatedAndLaundryStateChanged(
     }
 
+    /**
+     * Checks that in case of an attempt of processing order with wrong status (not {@link OrderStatus#SUBMITTED}),
+     * the exception is thrown and DB is not changed.
+     * Input: order with {@link OrderStatus#RESERVED}
+     * Output: error, because only order with {@link OrderStatus#SUBMITTED} can be processed (or change it's status
+     * to {@link OrderStatus#COMPLETED} using PUT /orders/{id}/status/completed
+     *
+     * @throws Exception if object cannot be serialized
+     */
+    @Test
+    @Sql(scripts = "/test-data/basic.sql")
+    @Transactional
+    public void testProcessCompletedOrder_orderIsReserved_orderIsNotProcessedDbIsNotChanged() throws Exception {
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(
+                EXISTING_LAUNDRY_ID,
+                INBOUND_LAUNDRY_WAITING_TIME,
+                INBOUND_LAUNDRY_STATE_VERSION);
+
+        OrderCompletedDto orderCompletedDto = new OrderCompletedDto(
+                EXISTING_RESERVED_ORDER_ID,
+                inboundLaundryStateDto,
+                DEFAULT_COMPLETION_TIME);
+
+        LaundryStateDto initialLaundryState = laundryStateService.getLaundryStateById(EXISTING_LAUNDRY_ID);
+        OrderDto initialOrder = orderService.getOrderById(EXISTING_RESERVED_ORDER_ID);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderCompletedDto);
+        try {
+            Exception mvcResult = mockMvc.perform(put("/orders/" + EXISTING_RESERVED_ORDER_ID + "/status/completed")
+                    .contentType(APPLICATION_JSON.toString())
+                    .content(orderSubmittedJson))
+                    .andReturn().getResolvedException();
+            if (mvcResult != null) {
+                throw mvcResult;
+            } else {
+                fail();
+            }
+        } catch (Exception e) {
+            LaundryStateDto updatedLaundryState = laundryStateService.getLaundryStateById(EXISTING_LAUNDRY_ID);
+            OrderDto updatedOrder = orderService.getOrderById(EXISTING_RESERVED_ORDER_ID);
+
+            assertEquals(initialLaundryState, updatedLaundryState);
+            assertEquals(initialOrder, updatedOrder);
+        }
+
+    }
+
+    @Test(expected = MethodArgumentNotValidException.class)
+    public void testProcessCompletedOrder_invalidQueueWaitingTime_orderValidationExceptionIsThrown() throws Exception {
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(
+                EXISTING_LAUNDRY_ID,
+                -100L,
+                INBOUND_LAUNDRY_STATE_VERSION);
+
+        OrderCompletedDto orderCompletedDto = new OrderCompletedDto(
+                EXISTING_SUBMITTED_ORDER_ID,
+                inboundLaundryStateDto,
+                DEFAULT_COMPLETION_TIME);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderCompletedDto);
+        MvcResult mvcResult = mockMvc.perform(put("/orders/666/status/completed")
```

Comment:
Do not use wrong URL parameter when it's not needed. You test another thing here